<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CreateServerDialog</class>
 <widget class="QDialog" name="CreateServerDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>538</width>
    <height>366</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Xonotic Server Details</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Server</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="input_preset">
       <property name="toolTip">
        <string>Server Preset</string>
       </property>
       <property name="whatsThis">
        <string>Selects a saved server or &quot;Create New&quot; to create a new server</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="formGroupBox">
     <property name="title">
      <string>Server Details</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Secure</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="input_host">
        <property name="toolTip">
         <string>Server host or address</string>
        </property>
        <property name="whatsThis">
         <string>Host name or IP address of the Xonotic server</string>
        </property>
        <property name="text">
         <string>localhost</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Server</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="input_port">
        <property name="toolTip">
         <string>Port Number</string>
        </property>
        <property name="whatsThis">
         <string>UPD Port number of the Xonotic server</string>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>26000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Port</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Password</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="input_secure">
        <property name="toolTip">
         <string>Rcon Secure Protocol</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rcon secure protocol, must match the value of the cvar &lt;span style=&quot; font-weight:600;&quot;&gt;rcon_secure&lt;/span&gt; in the Xonotic server.&lt;/p&gt;&lt;p&gt;&amp;quot;No&amp;quot; corresponds to &amp;quot;rcon_secure 0&amp;quot;, and it's the recommended option.&lt;/p&gt;&lt;p&gt;&amp;quot;Time Based&amp;quot; hashes the command based on the system time, it requires to have the system clock to be synchronized with the server clock.&lt;/p&gt;&lt;p&gt;&amp;quot;Challenge Based&amp;quot; asks the server for challenges, secure but not reliable.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <item>
         <property name="text">
          <string>No</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Time Based</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Challenge Based</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="input_password">
        <property name="toolTip">
         <string>Rcon Password</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Rcon password, it must match the cvar &lt;span style=&quot; font-weight:600;&quot;&gt;rcon_password&lt;/span&gt; as set on the Xonotic server&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="input_name">
        <property name="toolTip">
         <string>Name used to identify the server</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This name is used to recognize the server and as an identifier for the server presets. If omitted it will become server:port&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="button_delete">
          <property name="whatsThis">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Deletes the server with the given name&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>&amp;Delete</string>
          </property>
          <property name="icon">
           <iconset theme="edit-delete">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="button_save">
          <property name="whatsThis">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Save the current server setup under the given name&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="text">
           <string>&amp;Save</string>
          </property>
          <property name="icon">
           <iconset theme="document-save">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>input_preset</tabstop>
  <tabstop>input_host</tabstop>
  <tabstop>input_port</tabstop>
  <tabstop>input_password</tabstop>
  <tabstop>input_secure</tabstop>
  <tabstop>input_name</tabstop>
  <tabstop>button_save</tabstop>
  <tabstop>button_delete</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CreateServerDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>497</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CreateServerDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
